ui/src/routes/devices.$id.settings.macros.tsx

```diff
@@ -0,0 +1,1439 @@
+                          { keys: [], modifiers: [], delay: DEFAULT_DELAY }
+                        ],
+                      }));
+                      clearErrors();
+                    }}
+                    disabled={isMaxStepsReachedForNewMacro}
+                  >
+                    <LuPlus className="h-4 w-4" />
+                    <span>Add Step {isMaxStepsReachedForNewMacro && `(${MAX_STEPS_PER_MACRO} max)`}</span>
+                  </button>
+                </div>
+
+                <div className="mt-6 flex items-center justify-between border-t border-slate-200 pt-4 dark:border-slate-700">
+                  {showClearConfirm ? (
+                    <div className="flex items-center gap-2">
+                      <span className="text-sm text-slate-600 dark:text-slate-400">
+                        Cancel changes?
+                      </span>
+                      <Button
+                        size="SM"
+                        theme="danger"
+                        text="Yes"
+                        onClick={() => {
+                          resetNewMacro();
+                          setShowAddMacro(false);
+                          setShowClearConfirm(false);
+                        }}
+                      />
+                      <Button
+                        size="SM"
+                        theme="light"
+                        text="No"
+                        onClick={() => setShowClearConfirm(false)}
+                      />
+                    </div>
+                  ) : (
+                    <div className="flex gap-x-2">
+                      <Button
+                        size="SM"
+                        theme="primary"
+                        text={isSaving ? "Saving..." : "Save Macro"}
+                        onClick={handleAddMacro}
+                        disabled={isSaving}
+                      />
+                      <Button
+                        size="SM"
+                        theme="light"
+                        text="Cancel"
+                        LeadingIcon={LuX}
+                        onClick={() => {
+                          if (newMacro.name || newMacro.description || newMacro.steps?.some(s => s.keys?.length || s.modifiers?.length)) {
+                            setShowClearConfirm(true);
+                          } else {
+                            resetNewMacro();
+                            setShowAddMacro(false);
+                          }
+                        }}
+                      />
+                    </div>
+                  )}
+                </div>
+              </div>
+            </div>
+          </div>
+        )}
+        {macros.length > 0 && (
+          <div className="space-y-2">
+            
+            {macros.length === 0 ? (
+              <p className="text-center text-sm text-slate-500 dark:text-slate-400 py-4">
+                No macros created yet. Add your first macro above.
```

Comment:
This is never displayed.

Use EmptyCard instead. If there are no items, remove the CTA from the top header and show EmptyCard. Once items exist, display the “Add Macro” CTA in the header. Only one CTA should be shown at a time.

Reply:
cleaned up, thanks!